sanic/app.py

```diff
@@ -443,6 +443,11 @@ def dispatch(
         inline: bool = False,
         reverse: bool = False,
     ) -> Coroutine[Any, Any, Awaitable[Any]]:
+        if self.config.get("SKIP_TOUCHUP", False):
```

Comment:
I'd rather take the value of this and cache it on the `SignalRouter` instance. Then inside the `_dispatch` method we can check for the value _inside_ the exception block. Therefore we do not need to check this value on **every** dispatch of every event.

```
# ./signals.py
async def _dispatch(...):
    try:
        ...
    except Notfound as e:
        if fail_not_found and self.allow_failures:
            raise e
        ...
```

We could set it is inside `Sanic.signalize`. The other option that probably makes more sense is to tie it into the `Config` object with `_post_set` so that we are always syncing the config value back to the `SignalRouter` since we ultimately want that to be the public facing API.

Reply:
Been doing some work around integrating this suggestion. An issue I'm running into: the config _post_set() method does not have access to the `app` instance, so it cannot affect the app's or get the SignalRouter instance. And the Config object may not even be attached to an app when the config property is set.

I will however look into integrating the property into the signal router

Reply:
Ah, I see another problem. The value of `app.config.SKIP_TOUCHUP` is likely not set at the point in time that the SignalRouter is created.

Reply:
We could make a setter on `app`:
```
@property
    def skip_touchup(self):
        return self.config.SKIP_TOUCHUP

    @skip_touchup.setter
    def skip_touchup(self, value: bool):
        self.signal_router.allow_fail_builtin = not value
        self.config.SKIP_TOUCHUP = value
```

Reply:
I think we can also do it as a descriptor on the `Config` object. This is similar to what we did with `FALLBACK_ERROR_FORMAT` since we ultimately want the `Config` object to be "the source of truth".

Reply:
> I will however look into integrating the property into the signal router. The tricky part is it should only skip throwing an exception on the built-in signals, not for any user-defined signals.

Agreed

Reply:
Actually... as I think more and more about this, we only care about what the value is at startup time. So, it can be a regular config variable that we read inside `_startup` and pass to `signalize`. Any change to the config value later is irrelevant, and we just change the `signal_router` value at startup time.